init.lua

```diff
@@ -1,60 +1,89 @@
+local storage = minetest.get_mod_storage()
 local themename = ""
 
 if minetest.global_exists("dreambuilder_theme") then
 	themename = dreambuilder_theme.name.."_"
 end
 
-local player_hotbar_settings = {}
-local f = io.open(minetest.get_worldpath()..DIR_DELIM.."hotbar_settings","r")
-if f then
-	player_hotbar_settings = minetest.deserialize(f:read("*all"))
-	f:close()
+local base_img = themename.."gui_hb_bg_1.png"
+local imgref_len = string.len(base_img) + 8 -- accounts for the stuff in the string.format() below.
+local hotbar_size_default = 16
+
+local img = {}
+for i = 0, 31 do
+	img[i+1] = string.format(":%04i,0=%s", i*64, base_img)
 end
+local hb_img = table.concat(img)
 
 local function validate_size(s)
-	local size = s and tonumber(s) or 16
+	local size = s and tonumber(s) or hotbar_size_default
 	return math.floor(0.5 + math.max(1, math.min(size, 32)))
 end
 
-local hotbar_size_default = validate_size(minetest.settings:get("hotbar_size"))
+hotbar_size_default = validate_size(minetest.settings:get("hotbar_size"))
 
-local base_img = themename.."gui_hb_bg_1.png"
-local imgref_len = string.len(base_img) + 8 -- accounts for the stuff in the string.format() below.
+local function migrate_file2modstorage()
+	local path = minetest.get_worldpath()..DIR_DELIM.."hotbar_settings"
+	local f = io.open(path, "r")
+	if f then
+		local hotbar_sizes = minetest.deserialize(f:read("*all"))
+		f:close()
+		local counter = 0
+		for name, size in pairs(hotbar_sizes) do
+			if size ~= hotbar_size_default then
+				storage:set_int(name, tonumber(size))
+				counter = counter + 1
+			end
+		end
+		os.remove(path)
+		minetest.log("action", "[dreambuilder_hotbar] Migrated " .. counter .. " player hotbars to modstorage")
+	end
+end
 
-local img = {}
-for i = 0, 31 do
-	img[i+1] = string.format(":%04i,0=%s", i*64, base_img)
+local function get_hotbar_size(player)
+	local name = player:get_player_name()
+	local meta = player:get_meta()
+	local size = meta:get_int("hotbar_size")
+	if size == 0 then -- not present
+		size = storage:get_int(name)
+		if size ~= 0 then -- migrate
+			storage:set_string(name, "")
+			meta:set_int("hotbar_size", size)
+		end
+	end
+	if size == 0 then size = nil end
+	return size or hotbar_size_default
```

Comment:
Could simplified to this: (fine either way though)
```suggestion
	return size > 0 and size or hotbar_size_default
```